src/components/Editor.js

```diff
@@ -24,13 +24,14 @@ type Props = {
 };
 
 type State = {
-  code: string,
+  code: string
 };
 
 export default class Editor extends Component<Props, State> {
-  state = {
-    code: '',
-  };
+  constructor(props) {
+    super(props);
+    this.state = ((typeof (window.initialCode)) === 'undefined') ? { code: '' } : { code: window.initialCode };

```

Comment:
this can be reduced to `window.initialCode || ''`. Also let's keep the initial state so let's make it
```javascript
state = {
  code: window.initialCode || '',
};
```

Reply:
What's the difference between using constructor and the way you wrote it.

Reply:
Both ways do the same thing. The way it is written is the modern syntax for the constructor that only initializes the state.